src/main/kotlin/com/motycka/edu/lesson02/Functions.kt

```diff
@@ -1,30 +1,48 @@
 package com.motycka.edu.lesson02
 
 val coffeeOrders = mutableMapOf<Int, List<String>>()
+var nextOrderId = 1
 
-fun main() {
-    // You can write code here to try the functions
-    processOrder(listOf(ESPRESSO, CAPPUCCINO, CAPPUCCINO, AMERICANO), 20.0)
-    processOrder(listOf(ESPRESSO, FLAT_WHITE, AMERICANO), 10.0)
-    // processOrder(listOf(ESPRESSO, ESPRESSO, DOUBLE_ESPRESSO), 5.0) // will fail due to insufficient payment
+fun placerOrder(items: List<String>): Int {
+    val id = nextOrderId++
+    coffeeOrders[id] = items
+    println("Order placed: ID=$id, Items=$items")
+    return id
 }
 
-/* Implement the functions below */
+fun payOrder(orderId: Int): Double {
+    val items = coffeeOrders[orderId] ?: emptyList()
+    val total = items.sumOf { getPrice(it) }
+    println("Order ID $orderId total amount: $$total")
+    return total
+}
+
+fun completeOrder(orderId: Int) {
+    coffeeOrders.remove(orderId)
+    println("Order ID $orderId completed and removed from the system.")
+}
 
 fun processOrder(items: List<String>, payment: Double): Double {
-    val orderId = TODO("call placerOrder(items)")
-    val totalToPay = TODO("call payOrder(orderId)")
+    val orderId = placerOrder(items)
+    val totalToPay = payOrder(orderId)
+
+    if (payment < totalToPay) {
+        println("Insufficient payment! Payment: $$payment, required: $$totalToPay")
+        // You can handle this by throwing an exception or returning negative change
+        completeOrder(orderId)  // optional: cancel order on failure
+        return -1.0
```

Comment:
I would recommend throwing an error rather than return -1 because if you return -1 the caller will have to understand what -1 means. What would it mean for example in terms of response to the customer? Throwing an error is cleaner.